index.js

```diff
@@ -29,65 +29,44 @@ var addFrame = function (encoder, canvas, word, opts) {
 	encoder.addFrame(canvas);
 };
 
-var generateFileName = function (opts) {
-	return new Promise(function (resolve) {
-		var flag = true;
-		while (flag) {
-			if (opts.filename !== undefined && !pathExists.sync(opts.filename)) {
-				flag = false;
-				resolve(opts);
-			} else {
-				opts.filename = 'meme-' + randomInt(100, 99999) + '.gif';
-			}
-		}
-	});
+var generateFileName = function () {
+	var filename;
+	while (filename === undefined || pathExists.sync(filename)) {
+		filename = 'meme-' + randomInt(100, 99999) + '.gif';
+	}
+
+	return filename;
 };
 
 module.exports = function (text, opts) {
-	return new Promise(function (resolve, reject) {
-		if (typeof text !== 'string' || text === '') {
-			reject(new Error('Expected some string value'));
-		}
-
-		opts = opts || {};
-		opts.repeat = 0;
-		// 0 to repeat and 1 to not repeat
-		opts.quality = 10;
-		text = text.split(' ');
+	if (typeof text !== 'string' || text === '') {
+		throw new Error('Expected some string value');
+	}
 
-		if (opts.background === undefined) {
```

Comment:
Use `object-assign` for this. You are altering the users object + it's a lot cleaner then all these if statements.